src/index.ts

```diff
@@ -590,20 +602,28 @@ export default class Purchases {
 
   /**
    * Sets up Purchases with your API key and an app user id.
-   * @param {string} apiKey RevenueCat API Key. Needs to be a String
+   * @param {String} apiKey RevenueCat API Key. Needs to be a String
    * @param {String?} appUserID An optional unique id for identifying the user. Needs to be a string.
    * @param {Boolean?} observerMode An optional boolean. Set this to TRUE if you have your own IAP implementation and want to use only RevenueCat's backend. Default is FALSE.
+   * @param {String?} userDefaultsSuiteName An optional string. iOS only. Set this to use a specific NSUserDefaults suite for RevenueCat.
+   * This might be handy if you are deleting all NSUserDefaults in your app and leaving RevenueCat in a bad state.
    * @returns {Promise<void>} Returns when setup completes
    */
   public static setup(
     apiKey: string,
     appUserID?: string | null,
-    observerMode: boolean = false
+    observerMode: boolean = false,
+    userDefaultsSuiteName?: string
   ) {
-    if (typeof appUserID !== "undefined" && typeof appUserID !== "string") {
+    if (appUserID !== null && typeof appUserID !== "undefined" && typeof appUserID !== "string") {
```

Comment:
I realized people couldn't do something like:

`Purchases.setup("api_key", null, false, "suitename")`

We would throw an exception saying the `appUserID needs to be a string`, but `appUserID` can be null